ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/DdlCommandExec.java

```diff
@@ -52,31 +52,44 @@ public DdlCommandExec(final MutableMetaStore metaStore) {
     this.metaStore = Objects.requireNonNull(metaStore, "metaStore");
   }
 
+  public DdlCommandResult execute(
+      final String sql,
+      final DdlCommand ddlCommand,
+      final boolean withQuery,
+      final Set<SourceName> withQuerySources
+  ) {
+    return execute(sql, ddlCommand, withQuery, withQuerySources, false);
+  }
+
   /**
    * execute on metaStore
    */
   public DdlCommandResult execute(
       final String sql,
       final DdlCommand ddlCommand,
       final boolean withQuery,
-      final Set<SourceName> withQuerySources
+      final Set<SourceName> withQuerySources,
+      final boolean restoreInProgress
   ) {
-    return new Executor(sql, withQuery, withQuerySources).execute(ddlCommand);
+    return new Executor(sql, withQuery, withQuerySources, restoreInProgress).execute(ddlCommand);
   }
 
   private final class Executor implements io.confluent.ksql.execution.ddl.commands.Executor {
     private final String sql;
     private final boolean withQuery;
     private final Set<SourceName> withQuerySources;
+    private final boolean restoreInProgress;
 
     private Executor(
         final String sql,
         final boolean withQuery,
-        final Set<SourceName> withQuerySources
+        final Set<SourceName> withQuerySources,
+        final boolean restoreInProgress
     ) {
       this.sql = Objects.requireNonNull(sql, "sql");
       this.withQuery = withQuery;
       this.withQuerySources = Objects.requireNonNull(withQuerySources, "withQuerySources");
+      this.restoreInProgress = Objects.requireNonNull(restoreInProgress, "restoreInProgress");
```

Comment:
nit: As a primitive, boolean cannot be null.  `withQuery` isn't checked.

Reply:
True. Done